node/cloud_functions/basicBot/index.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Description: A basic Chat Bot setup, to be used with Google Cloud Functions.
+ */
+
+var BOT_NAME = "Basicbot";
+
+exports.basicBot = (req, res) => {
+  switch (req.body.type) {
+    case "ADDED_TO_SPACE":
+      res.send(handleAddToSpace(req.body));
+      break;
+    case "MESSAGE":
+      res.send(handleMessage(req.body));
+      break;
+    case "CARD_CLICKED":
+      res.send(handleCardClick(req.body));
+      break;
+  }
+};
+
+/**
+ * Handles interactions when when the Bot is added to the Room
+ * @param {Object} body
+ * @returns {Object} returns a Card with widgets containing key value pairs
+ */
+function handleAddToSpace(body) {
+  return getHelp();
+}
+
+/**
+ * Handles Type of MESSAGE, cleans the text to find what search text was used and returns proper response
+ * @param {Object} body , the request body object, here is deferenced to more easily use the text string
+ * @returns {Object} Returns a Card object with multiple images and buttons
+ */
+function handleMessage({ text, ...body }) {
+  switch (cleanText(text)) {
+    case "i love dogs":
+      return getInteractiveCard();
+      break;
+  }
+}
+
+/**
+ * Handles the interaction when a card is clicked. Returns a single image
+ * @param {Object} body
+ * @returns {Object} returns a card object with the proper actionResponse to update the existing comment
+ */
+function handleCardClick(body) {
+  let { cards } = getSingleImage(body.action.parameters[0].value);
+  return {
+    actionResponse: {
+      type: "UPDATE_MESSAGE"
+    },
+    cards
+  };
+}
+
+/**
+ * Removed bot name, extra spaces, and returns back the important part of the text sent
+ * @param {String} text
+ * @return {String}
+ */
+function cleanText(text) {
+  return text
+    .toLowerCase()
+    .replace(`@${BOTNAME.toLowerCase()}`, "")
+    .split(" ")
+    .filter(char => char !== "")
+    .join(" ");
+}
+
+/**
+ * Returns a static interactive card, you can map over arrays to populate something similar
+ * @returns {Object}
+ */
+function getInteractiveCard() {
+  let imageUrls = [
+    "http://www.freepngimg.com/download/dog/9-dog-png-image-picture-download-dogs.png",
```

Comment:
I think these photos should be different.

Reply:
Changed to 3 unique urls.